### PR TITLE
add check and tests for if an optional value is set as an empty string

### LINKED ISSUE #246: [Add Check]: Missing values should be `None`, not empty string
### What would you like to see added to the NWBInspector?

Sometimes users are missing optional values, such as the NWBFile.experimenter, and they use an empty string as a placeholder instead of just omitting this value by setting it to `None`. `None` is the better approach. We should add this to the best practices and add checks for it.

### Pseudocode or other logic of the check

_No response_

### Do you have any interest in helping implement the check function?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this check was not already [requested](https://github.com/neurodatawithoutborders/nwbinspector/issues)?

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

fix #246 

What was the reasoning behind this change? Please explain the changes briefly.

## How to test the behavior?
```
Show here how to reproduce the new behavior (can be a bug fix or a new feature)
```

## Checklist

- [ ] Have you ensured the PR description clearly describes the problem and solutions?
- [ ] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/neurodatawithoutborders/nwbinspector/pulls) for the same change?
- [ ] Is your code contribution compliant with `black` format? If not, simply `pip install black` and run `black .` inside your fork.
- [ ] If this PR fixes an issue, is the first line of the PR description `fix #XX` where `XX` is the issue number?

### CODE REVIEW

File: nwbinspector/checks/nwb_containers.py [hunk 1]
```diff
@@ -73,5 +73,5 @@ def check_empty_string_for_optional_attribute(nwb_container: NWBContainer):
     for field in fields:
         yield InspectorMessage(
             message=f'The attribute "{field}" is optional and you have supplied an empty string. Improve my omitting '
-                    f'this attribute (in MatNWB or PyNWB) or entering as None (in PyNWB)'
+            f"this attribute (in MatNWB or PyNWB) or entering as None (in PyNWB)"
```
Comment (CodyCBakerPhD) [inline nwbinspector/checks/nwb_containers.py]: Minor; This level doesn't have to be an fstring to concatenate with the one above it

File: nwbinspector/checks/nwb_containers.py [hunk 2]
```diff
@@ -73,5 +73,5 @@ def check_empty_string_for_optional_attribute(nwb_container: NWBContainer):
     for field in fields:
```
Comment (CodyCBakerPhD) [inline nwbinspector/checks/nwb_containers.py:73]: The logic for `optional_attrs` is a bit to unpack - is there a better variable name than `_d` for explaining what we're iterating over?

And just to confirm, `"args"` is all we need from the `__docval__` attached to the `__init__` of some class? No `"kwargs"` alternatives?

And the logic for detecting optionality is the presence of a `"default"` value, and if there is one we check that it's set to `None` instead of some pre-defined value? Just curious, how does the `None` logic equate to a file written in MATLAB? Does the attribute simply not show up attached to the instance of the class when read back in through PyNWB?

Comment (bendichter) [reply nwbinspector/checks/nwb_containers.py:73]: > The logic for optional_attrs is a bit to unpack - is there a better variable name than _d for explaining what we're iterating over?

ok, I renamed `_d` to `arg`. 

> And just to confirm, "args" is all we need from the __docval__ attached to the __init__ of some class? No "kwargs" alternatives?

docval just has everything in "args". There is no "kwargs" field.

> how does the None logic equate to a file written in MATLAB 

In MatNWB, all attributes are written with key,value pairs, e.g. 

```matlab
NwbFile("session_start_time", now(), "session_description", "this is my description")
```

To omit an attribute, you just don't provide the key, value pair. Omitted values are read back as `[]`.

File: nwbinspector/checks/nwb_containers.py [hunk 3]
```diff
@@ -67,11 +67,11 @@ def check_empty_string_for_optional_attribute(nwb_container: NWBContainer):
...
-    docval = type(nwb_container).__init__.__docval__["args"]
-    optional_attrs = [_d["name"] for _d in docval if "default" in _d and _d["default"] is None]
+    docval_args = type(nwb_container).__init__.__docval__["args"]
+    optional_attrs = [arg["name"] for arg in docval_args if "default" in arg and arg["default"] is None]
     fields = [attr for attr in optional_attrs if getattr(nwb_container, attr) == ""]
     for field in fields:
         yield InspectorMessage(
             message=f'The attribute "{field}" is optional and you have supplied an empty string. Improve my omitting '
-                    f'this attribute (in MatNWB or PyNWB) or entering as None (in PyNWB)'
+            f"this attribute (in MatNWB or PyNWB) or entering as None (in PyNWB)"
```
Comment (bendichter) [inline nwbinspector/checks/nwb_containers.py]:
```suggestion
            "this attribute (in MatNWB or PyNWB) or entering as None (in PyNWB)"
```

File: nwbinspector/checks/nwb_containers.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply nwbinspector/checks/nwb_containers.py:73]: Oky doky, sounds good then

**Review by CodyCBakerPhD — APPROVED** [review]
